### PR TITLE
doc: config_and_build: gather concepts on one page

### PR DESCRIPTION
**Pull request by greg-fer**
Edited the overview page to gather conceptual information on one page. This removes some duplication and allows to expand procedures with more content in a future PR.
NCSDK-22324.

### CODE REVIEW

**Comment by greg-fer** [comment]
Local output:
image
image
image

**Comment by greg-fer** [comment]
The preview in the comment doesn't have the includes from Zephyr. The Hardware and Software sections should look like follows:
image

File: doc/nrf/config_and_build/config_and_build_system.rst [hunk 1]
```diff
@@ -9,28 +9,159 @@ Build and configuration system
...
+You can then edit them according to your needs (see :ref:`modifying`).
+
+When you start building, a CMake build is executed in two stages: configuration phase and building phase.
+
+.. _configuration_system_overview_config:
+
+Configuration phase
+===================
+
+During this phase, CMake executes build scripts from :file:`CMakeLists.txt` and gathers configuration from different sources to generate the final build scripts at the application source directory and create a model of the build for the specified build target.
```
Comment (tejlmand) [inline doc/nrf/config_and_build/config_and_build_system.rst:62]: > to generate the final build scripts at the application source directory

not sure exactly what you try to describe but sounds wrong to me.
A benefit of CMake is exactly that you avoid in-source builds and can place the build folder anywhere you like, for example: `cmake -B/foo/build sample/bar` will place the generated build files in `/foo/build` which is different from `sample/bar` which holds the source files.

**Review by tejlmand — COMMENTED** [review]
a nice improvement.

Noticed some minor nits, nothing which destroys the meaning but still minor inaccuracies that would be nice to get corrected.

I do notice most text are moved from other location, but now would be good opportunity to correct them.

File: doc/nrf/config_and_build/config_and_build_system.rst [hunk 2]
```diff
@@ -9,28 +9,159 @@ Build and configuration system
...
+
+.. ncs-include:: build/cmake/index.rst
+   :docset: zephyr
+   :dedent: 3
+   :start-after: Kconfig
+   :end-before: Information from devicetree is available to Kconfig,
+
+Information from devicetree is available to Kconfig, through the functions defined in :file:`zephyr/scripts/kconfig/kconfigfunctions.py`.
+
+The single :file:`.config` file in the :file:`<build_dir>/zephyr/` directory describes the entire software configuration of the constructed binary.
```
[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/config_and_build/config_and_build_system.rst:113]: Rephrased to:
> The :file:`.config` file in the :file:`<build_dir>/zephyr/` directory describes most of the software configuration of the constructed binary.
Some subsystems can use their own configuration files.

File: doc/nrf/config_and_build/config_and_build_system.rst (hunk 1 above)
Comment (greg-fer) [reply doc/nrf/config_and_build/config_and_build_system.rst:62]: Rephrased to:
> During this phase, CMake executes build scripts from :file:`CMakeLists.txt` and gathers configuration from different sources to generate the final build scripts and create a model of the build for the specified build target.

**Review by tejlmand — APPROVED** [review]

**Review by DeidreCasey — APPROVED** [review]

**Pull request merged**